### PR TITLE
Test pr 1707

### PR DESCRIPTION
**Pull request by sfc-gh-aalam**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes #NNNN

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am modifying authorization mechanisms
   - [ ] I am adding new credentials
   - [ ] I am modifying OCSP code
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Test PR for #1707 .

### CODE REVIEW

File: src/snowflake/connector/pandas_tools.py [hunk 1]
```diff
@@ -280,6 +294,27 @@ def write_pandas(
...
         )
 
+    # use_logical_type should be True when dataframe contains datetimes with timezone.
+    # https://github.com/snowflakedb/snowflake-connector-python/issues/1687
+    if not use_logical_type and any(
+        [pandas.api.types.is_datetime64tz_dtype(df[c]) for c in df.columns]
+    ):
+        warnings.warn(
+            "Dataframe contains a datetime with timezone column, but "
+            f"'use_logical_type = {use_logical_type}'. This can result in dateimes "
```
Comment (sfc-gh-mkeller) [inline src/snowflake/connector/pandas_tools.py:304]: There's a shortcut for this starting with Python 3.8 https://docs.python.org/3/whatsnew/3.8.html#f-strings-support-for-self-documenting-expressions-and-debugging

File: test/integ/pandas/test_pandas_tools.py [hunk 2]
```diff
@@ -417,6 +417,54 @@ def test_write_pandas_create_temp_table_deprecation_warning(
             cnx.execute_string(drop_sql)
 
 
+@pytest.mark.parametrize("use_logical_type", [None, True, False])
+def test_write_pandas_use_logical_type(
```
Comment (sfc-gh-mkeller) [inline test/integ/pandas/test_pandas_tools.py:421]: Please add a unit test for verifying the warning you emit

File: test/integ/pandas/test_pandas_tools.py [hunk 3]
```diff
@@ -417,6 +417,54 @@ def test_write_pandas_create_temp_table_deprecation_warning(
             cnx.execute_string(drop_sql)
 
 
+@pytest.mark.parametrize("use_logical_type", [None, True, False])
```
Comment (sfc-gh-mkeller) [inline test/integ/pandas/test_pandas_tools.py:420]: Could we add unit tests for for all possible combinations and only run a single integration test?

**Review by sfc-gh-mkeller — CHANGES_REQUESTED** [review]

File: src/snowflake/connector/pandas_tools.py (hunk 1 above)
Comment (dvorst) [reply src/snowflake/connector/pandas_tools.py:304]: [fixed](https://github.com/snowflakedb/snowflake-connector-python/pull/1707/commits/fb84282b04e66f7decb7732b37e2001c2b0b3009)
also [synced with upstream](https://github.com/snowflakedb/snowflake-connector-python/pull/1707/commits/0798b1fdab15dd43930e717db47ca506cb7ffb6a)

File: test/integ/pandas/test_pandas_tools.py (hunk 2 above)
Comment (dvorst) [reply test/integ/pandas/test_pandas_tools.py:421]: The warning is verified in the same test at line 462

Comment (dvorst) [reply test/integ/pandas/test_pandas_tools.py:421]: The warning is tested in the same test at line 462. I could split off testing the warning from the integration test if this is preferred. Other warnings, like the temp_table depreciation warning, are currently also tested during the integration tests, see "test_write_pandas_create_temp_table_deprecation_warning".

File: test/integ/pandas/test_pandas_tools.py (hunk 3 above)
Comment (dvorst) [reply test/integ/pandas/test_pandas_tools.py:420]: use_logical_type is used to create sql_use_logical_type in write_pandas, which is directly passed to a SQL command (line 420 of pandas_tools.py). Testing only 1 parameter value and mocking the other 2 values, would therefore not cover the entire implementation.

**Comment by dvorst** [comment]
I see the tests seem to fail due to connection issues, I [synced with upstream](https://github.com/snowflakedb/snowflake-connector-python/pull/1707/commits/0798b1fdab15dd43930e717db47ca506cb7ffb6a), might this solve the issues at hand? Let me know if there is anything I can do from my side.

File: test/integ/pandas/test_pandas_tools.py (hunk 3 above)
Comment (sfc-gh-mkeller) [reply test/integ/pandas/test_pandas_tools.py:420]: > Testing only 1 parameter value and mocking the other 2 values, would therefore not cover the entire implementation.

Right, unit testing would ensure that we have correctness and cover everything, but if we had an integration test for each of the `write_pandas` flags then the tests would run for twice as long at least.

**Review by sfc-gh-mkeller — APPROVED** [review]
:ship: :ship:

**Pull request merged**